### DISCUSSION TITLE
Enhancement: Let performOn use arg 1 as target of arg2

### DISCUSSION THREAD

**Discussion by BobRay** [Codeception/Codeception#6197] [Ideas]
#### What are you trying to achieve?
I would like to use performOn in this way:

`$I->performOn("#orphans-{$nameLower}s-reload", 'click'), 5);`

It seems like it should be possible for performOn to use arg1 as the target and arg2 as the action if arg2 is not an array.

It's a common use case and would prevent having to specify the target twice.

**Comment by edno** [comment]
I might miss the point, but if you've a single action, then why not using directly the corresponding method ? 
Like `$I->click("#orphans-{$nameLower}s-reload")`

**Comment by BobRay** [comment]
Because click() doesn't wait for the element. performOn() does.

What I'm suggesting is that this code:

`$I->performOn($el, ['click' => $el]);`

Could be shortened to this shorter, more convenient, and more readable code, with a small change in the Codeception performOn method:

`$I->performOn($el, 'click');`

I suspect that this shorthand method would be used a lot, since waiting for an element, then clicking on it happens a lot in acceptance testing.
